gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -620,7 +620,10 @@ private boolean shouldStickToTheSameNode(int containerExitStatus) {
    */
   protected void handleContainerCompletion(ContainerStatus containerStatus) {
     Map.Entry<Container, String> completedContainerEntry = this.containerMap.remove(containerStatus.getContainerId());
-
+    if (completedContainerEntry == null) {
```

Comment:
In what kind of scenarios will it happen, when you say " we don't maintain this container " ?

Reply:
We have met some NPE when the container completion happens. And we alway add the containerId to the map when one container is allocated and remove the id only inside this method. So we assume that there may be chances that for one completed container, this method will be called more than once. And in that scenario (the method has already be called once to handle this completion), we can say we no longer maintain this container.